python/cudf/cudf/_lib/csv.pyx

```diff
@@ -365,6 +365,7 @@ def read_csv(
     if index_col is not None and index_col is not False:
         if isinstance(index_col, int):
             df = df.set_index(df._data.get_by_index(index_col).names[0])
```

Comment:
So the significance of an `int` being supplied here is to pick the column at that index and name that index with the same number..for example:

```python
buffer = "a,1,2\n3,4,5\n6,7,8"
names = ["int1", "int2", "int3"]


cu_df = read_csv(StringIO(buffer), header=None, index_col=0)
pd_df = pd.read_csv(StringIO(buffer), header=None, index_col=0)

cu_df.index
StringIndex(['a' '3' '6'], dtype='object', name=0) # This is now fixed, without the fix it would result in name='a'

pd_df.index
Index(['a', '3', '6'], dtype='object', name=0)
```

Reply:
But if names is passed, then it should take name from names and not int
```
>>> pd_df = pd.read_csv(StringIO(buffer), header=None, index_col=0, names=names)
>>> pd_df.index
Index(['a', '3', '6'], dtype='object', name='int1')
```

Reply:
Thanks for the catch, fixed it 👍